xtr/src/main.rs

```diff
@@ -232,6 +281,12 @@ fn main() -> Result<(), Error> {
             .value_of("charset")
             .expect("expected charset to have a default value")
             .to_owned(),
+        add_location: AddLocation::from_str(
+            matches
+                .value_of("add-location")
+                .expect("expected add-location to have a default value"),
+        )
+        .expect("expected add-location to be a valid value"),
```

Comment:
i wonder if it should not be `?` here to report the error.  but since this should be impossible, i guess panic is fine.

Reply:
yeah I was kind of thinking this situation shouldn't occur due to the argument checking in clap, so a panic would be okay.

Reply:
oops, sorry, I mean it shouldn't occur because all variants of `AddLocation` should be covered here, and if they aren't it would be nice to know about it